### PR TITLE
Added custom build section

### PR DESCRIPTION
**Pull request by joshuacox**
I made the other one unclean as it was from marketplace, and was conflicting in places and got messy, here is a clean PR, ignore #269

### CODE REVIEW

**Comment by jshimko** [comment]
Thanks so much for taking the time to contribute this @joshuacox!  (and for helping people with all of this stuff in Gitter!  It's always great to have someone else helping with Docker/deployment questions!)

I just have a few quick notes on this PR...

First, the `reaction deploy` command definitely shouldn't be in this section of docs.  That's a command for the managed platform that won't work without having a Reaction account, being logged in, and having an app set up on our platform already.  As written in this PR, that command will do nothing other than throw an error about not being logged in.  You can see the full process of using the platform here...

https://docs.reactioncommerce.com/reaction-docs/master/deploying-reaction-on-managed
(which is also where any docs related to that should be)

But just to be clear, that's currently an invite-only process (for the near future) and all of those commands that happen before `reaction deploy` are required steps for that deploy command to work.  So, we'll need to remove that whole section from this PR.

The other thing I wanted to mention is the two last sections for "Private Registry" and "manual export/import".  I feel like those kind of fall into the "advanced Docker usage" domain and they're not really typical/required use cases for the majority of people that would need to read these docs.  They're totally valid use cases and your docs are 100% correct, but I generally try to only include steps that are absolutely required when adding things to the deployment documentation.  I only say that because a few months ago I realized that a lot of people were getting confused/overwhelmed/stuck when trying to figure out how to go through the basic steps of Docker image deployment and I realized that I needed to trim down the detail to the bare minimum so it would feel a little more digestible.  Anyone that wants/needs to do those extra advanced steps probably 1) already knows how to or 2) should be referred to the Docker docs for the current best practices (which is often a moving target with Docker).

So in short, we need to remove the `reaction deploy` section, but I also think we should remove any bits that aren't absolutely necessary for creating a custom Docker image (custom registry and tar import/export). 

Make sense?

Again, thanks so much for the contribution!

**Comment by joshuacox** [comment]
You bet, perhaps should I move the latter bits to a separate page?

**Comment by machikoyasuda** [comment]
Thanks @joshuacox! 🎊
